doc/array.md

```diff
@@ -73,5 +84,73 @@ produces a panic error message if the index `i` is out of bounds. The latter ret
 #eval #['a', 'b', 'c'][5]?
 -- none
 #eval #['a', 'b', 'c'][1]!
--- 'b!
+-- 'b'
+```
+
+## Concatenation
+
+You can push a new item to the end of an array:
+```lean
+#eval Array.push #[1, 2, 3] 7
```

Comment:
I think we should emphasize the dot-notation in the manual. We rarely write `Array.push`. In the example above, we would write `#[1, 2, 3].push 7`.

Reply:
Done -- both here and in list.md